### DISCUSSION TITLE
Add fast transfer rate for the Pico/RP2040

### DISCUSSION THREAD

**Discussion by zapta** [Bodmer/TFT_eSPI#1227] [General]
Hello Bodmer,

The Pico/RP2040 contains Programmable IO (PIO) units that allow a very fast data transfer to the TFT with 8 or 16 bits parallel paths. For example, the proof of concept in the repository below uses a small PIO program and a 8 bit parallel path transfer 30MB/s or 15M pixels/sec  This an equivalent of > 240Mbps for a SPI interface.

https://github.com/zapta/pio_tft/blob/main/platformio/src/pio_tft.cpp

Is this something that you would like to add to TFT_eSPI? If so, if you can specified the expected API from the low level PIO driver, I can implement it. The current version of the API is at https://github.com/zapta/pio_tft/blob/main/platformio/src/pio_tft.h

The PIO program is at https://github.com/zapta/pio_tft/blob/main/platformio/src/pio_tft.pio . The assignment of the DB0-7, WR, DC, RST, BL, and CS pins is arbitrary, the only requirements is that DB0-DB7 will be consecutive RP2040 gpio pins.

**Comment by Bodmer** [comment]
Hello Zapta,
Thank you for the PIO code, I have incorporated it into a test branch and it is working well:
https://github.com/Bodmer/TFT_eSPI/tree/RP2040_8bit_parallel
 Typical results for the "graphicstest" sketch on a 320 x 480 ILI9488 screen:
```
Benchmark,                Time (microseconds)
Screen fill (5 times),    49711  (100.58 fps)
Text,                     7893
Lines,                    83295
Horiz/Vert Lines,         4349
Rectangles (outline),     2724
Rectangles (filled),      121211
Circles (filled),         29576
Circles (outline),        28686
Triangles (outline),      18275
Triangles (filled),       53626
Rounded rects (outline),  11216
Rounded rects (filled),   126054
Total = 536616us
Total = 0.5366s
```

**Comment by zapta** [comment]
Great. Thanks Bodmer!
